src/main/java/de/tum/in/www1/artemis/security/JiraAuthenticationProvider.java

```diff
@@ -97,24 +102,27 @@ public Authentication authenticate(Authentication authentication) throws Authent
         return new UsernamePasswordAuthenticationToken(user.getLogin(), user.getPassword(), grantedAuthorities);
     }
 
-    /**
-     * Gets or creates the user object for an JIRA user.
-     *
-     * @param authentication the Spring authentication object which includes the username and password
-     * @param skipPasswordCheck whether the password check on JIRA should be skipped
-     * @return
-     */
     @Override
+    public User getOrCreateUser(Authentication authentication, String firstName, String lastName, String email, Boolean skipPasswordCheck) {
+        // NOTE: firstName, lastName, email is not needed in this case since we always get these values from Jira
+        return getOrCreateUser(authentication, skipPasswordCheck);
+    }
+
     @SuppressWarnings("unchecked")
-    public User getOrCreateUser(Authentication authentication, Boolean skipPasswordCheck) {
+    private User getOrCreateUser(Authentication authentication, Boolean skipPasswordCheck) {
         String username = authentication.getName().toLowerCase();
         String password = authentication.getCredentials().toString();
-        HttpEntity<Principal> entity = new HttpEntity<>(
-                !skipPasswordCheck ? HeaderUtil.createAuthorization(username, password) : HeaderUtil.createAuthorization(JIRA_USER, JIRA_PASSWORD));
-        RestTemplate restTemplate = new RestTemplate();
         ResponseEntity<Map> authenticationResponse = null;
         try {
-            authenticationResponse = restTemplate.exchange(JIRA_URL + "/rest/api/2/user?username=" + username + "&expand=groups", HttpMethod.GET, entity, Map.class);
+            final var path = JIRA_URL + "/rest/api/2/user?username=" + username + "&expand=groups";
+            final HttpEntity<Principal> entity;
+            if (skipPasswordCheck) {
+                authenticationResponse = restTemplate.exchange(path, HttpMethod.GET, null, Map.class);
```

Comment:
this code does not have the same logic as before.
`entity  = new HttpEntity<>(HeaderUtil.createAuthorization(JIRA_USER, JIRA_PASSWORD)):` is missing. I guess this is the case, because the REST template already includes this authorization. I wonder however what happens with the REST template if the `else` branch is invoked. Does that change the default authorization?

Also, if the code is correct, you could move `entity` into the else branch completely.

Reply:
I guess some comments would be helpful here as well!

Reply:
Yes, the authorization gets injected into the `RestTemplate` (Artemis Admin auth). In the else case, we use the user's credentials, because we don't skip the password check. I added some comments to clarify this.